### DISCUSSION TITLE
Avoid whitespace changes for arrays in automatic config migrations

### DISCUSSION THREAD

**Discussion by JuJup** [renovatebot/renovate#30786] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I recently got a config migration PR from Renovate, which was correctly migrating some fields. However, it added whitespaces to all of my arrays.

Example:
 ```
Before:
{
      "matchFileNames": ["some-dir/**"],
      "matchDatasources": ["maven"],
      "matchDepTypes": ["dependencies"],
      "excludePackagePrefixes": [ "some-package:"],
},

After:
{
      "matchFileNames": [
        "some-dir/**"
      ],
      "matchDatasources": [
        "maven"
      ],
      "matchDepTypes": [
        "dependencies"
      ],
      "matchPackageNames": [
        "!some-package:{/,}**"
      ]
},
```

As seen in this example, `excludePackagePrefixes` was successfully migrated, but it also introduced a lot of unnecessary changes. While I prefer the inline arrays if it's a single entry it makes it also harder in the PR to see what really has changed.

I found [this Issue](https://github.com/renovatebot/renovate/issues/16383) where @rarkins explained the same problem. There seems to be a solution with prettier but I can't really get my head around that. When I run the validator locally, it doesn't shatter the JSON and when I run format on the file in VS Code it also doesn't change something.
I don't have an explicit `.prettierrc`, but even with adding this I couldn't find a solution to my problem.

Can someone point me in the right direction?

**Comment by HonkingGoose** [comment, marked as answer]
We want to make "getting config migration PRs from Renovate" the default behavior in a new major version (see #16359). But before we can make it the default, we need to stop/reduce the amount of whitespace changes.

I think this issue is most relevant here, as that is about using a different JSON parser, which may do better with whitespace changes?

- #27972

**Reply by JuJup** [reply]
Ok so I'll wait for that ticket and the next major version :) 
Thank you
